geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloFeatureStore.scala

```diff
@@ -50,6 +52,8 @@ class AccumuloFeatureStore(val dataStore: AccumuloDataStore, val featureName: Na
         val feature = iter.next()
         val newFeature = fw.next()
 
+        minMaxVisitorO.foreach ( _.visit(feature) )
```

Comment:
A feature collection can have an underlying iterator which is consumed by the first traversal.

If we want to save on the 'option check', let's build up a little lambdas for the two cases and then call it in the inner loop.
```
val write: SimpleFeature => Unit = if(dtg) { feature => updateBounds(feature); write(feature) } else write(feature)
while ( iter.hasNext) { 
  write(iter.next) 
 }
```